server/asset/btc/btc.go

```diff
@@ -493,6 +493,18 @@ func (btc *Backend) FeeRate(_ context.Context) (uint64, error) {
 	return btc.estimateFee(btc.node)
 }
 
+// SupportsDynamicTxFee returns true if the tx fee for this asset adjusts based
+// on market conditions
+func (*Backend) SupportsDynamicTxFee() bool {
+	return false
+}
```

Comment:
Wouldn't need to implement this here and in **dcr** if you made `SupportsDynamicTxFee` part of an optional interface like we're doing with the client.

Reply:
Actually, I'm now thinking this might not work like I expected, since it's just a boolean return value. So we'd verify that the interface is a DynamicFeeSupporter or whatever, and then still check the boolean? What we really need is something akin to WalletInfo, but on the server side. I'm okay with this as is too

Reply:
Yeah, it works as is.  @martonp can try some stuff out I guess, but I can see the optional method being `SwapFeeRate`, which would make `feeFetcher` unaware of the whole dynamic<->max_fee relationship.  Maybe we like that dynamic/max concept front and center?

Specifically, changing

```go
// SwapFeeRate returns the tx fee that needs to be used to initiate a swap.
// This fee will be the max fee rate if the asset supports dynamic tx fees,
// and otherwise it will be the current market fee rate.
func (f *feeFetcher) SwapFeeRate(ctx context.Context) uint64 {
	if f.Backend.SupportsDynamicTxFee() {
		return f.MaxFeeRate()
	}
	return f.FeeRate(ctx)
}
```

into

```go
// SwapFeeRate returns the tx fee that needs to be used to initiate a swap.
// This will use the Backend's SwapFeeRate method if it exists, otherwise
// it will request the fee rate based on current network conditions.
func (f *feeFetcher) SwapFeeRate(ctx context.Context) uint64 {
	if fr, ok := f.Backend.(asset.SwapFeeRater); ok {
		return fr.SwapFeeRate()
	}
	return f.FeeRate(ctx)
}
```

The question is do we hide the whole dynamic-must-use-max concept in the Backend's methods or do we apply that notion at the higher levels like done currently in this PR?

Reply:
This idea's not bad, maybe there will be another coin which requires some other technique.

Reply:
Ahh, that's right.  Max fee rate is a market/swapper concept, so just be done higher up.  Check.